eng/tools/generator/cmd/v2/automation/automationCmd.go

```diff
@@ -104,107 +107,41 @@ func (ctx *automationContext) generate(input *pipeline.GenerateInput) (*pipeline
-			if err != nil {
-				return nil, fmt.Errorf("failed to parse tspconfig.yaml: %+v", err)
-			}
-
-			if ok := tsc.ExistEmitOption(string(typespec.TypeSpec_GO)); ok {
-				log.Printf("Start to process typespec project: %s", tspProjectFolder)
-				existTypeSpecProject = true
-				generateCtx := common.GenerateContext{
-					SDKPath:        sdkRepo.Root(),
-					SDKRepo:        &sdkRepo,
-					SpecPath:       ctx.specRoot,
-					SpecCommitHash: ctx.commitHash,
-					SpecRepoURL:    input.RepoHTTPSURL,
-					TypeSpecConfig: tsc,
-				}
-
-				module, err := tsc.GetModuleName()
-				if err != nil {
-					return nil, err
-				}
-
-				namespaceResult, err := generateCtx.GenerateForTypeSpec(&common.GenerateParam{
-					RPName:              module[0],
-					NamespaceName:       module[1],
-					SkipGenerateExample: true,
-					GoVersion:           ctx.goVersion,
-				})
-				if err != nil {
-					errorBuilder.add(err)
-					continue
-				} else {
-					content := namespaceResult.ChangelogMD
-					breaking := namespaceResult.Changelog.HasBreakingChanges()
-					breakingChangeItems := namespaceResult.Changelog.GetBreakingChangeItems()
-
-					srcFolder := filepath.Join(sdkRepo.Root(), "sdk", "resourcemanager", namespaceResult.RPName, namespaceResult.PackageName)
-					apiViewArtifact := filepath.Join(sdkRepo.Root(), "sdk", "resourcemanager", namespaceResult.RPName, namespaceResult.PackageName+".gosource")
-					err := zipDirectory(srcFolder, apiViewArtifact)
-					if err != nil {
-						fmt.Println(err)
-					}
-
-					results = append(results, pipeline.PackageResult{
-						Version:         namespaceResult.Version,
-						PackageName:     fmt.Sprintf("sdk/resourcemanager/%s/%s", namespaceResult.RPName, namespaceResult.PackageName),
-						Path:            []string{fmt.Sprintf("sdk/resourcemanager/%s/%s", namespaceResult.RPName, namespaceResult.PackageName)},
-						PackageFolder:   fmt.Sprintf("sdk/resourcemanager/%s/%s", namespaceResult.RPName, namespaceResult.PackageName),
-						TypespecProject: []string{tspProjectFolder},
-						Changelog: &pipeline.Changelog{
-							Content:             &content,
-							HasBreakingChange:   &breaking,
-							BreakingChangeItems: &breakingChangeItems,
-						},
-						APIViewArtifact: fmt.Sprintf("sdk/resourcemanager/%s/%s", namespaceResult.RPName, namespaceResult.PackageName+".gosource"),
-						Language:        "Go",
-					})
-
-					log.Printf("Finish to process typespec file: %s", tspconfigPath)
-				}
-			}
-		}
-	}
-
-	if !existTypeSpecProject {
-		// autorest
-		if input.RelatedReadmeMdFile != "" {
-			input.RelatedReadmeMdFiles = append(input.RelatedReadmeMdFiles, input.RelatedReadmeMdFile)
+	for _, tspProjectFolder := range input.RelatedTypeSpecProjectFolder {
+		tspconfigPath := filepath.Join(input.SpecFolder, tspProjectFolder, "tspconfig.yaml")
+		tsc, err := typespec.ParseTypeSpecConfig(tspconfigPath)
+		if err != nil {
+			errorBuilder.add(fmt.Errorf("failed to parse %s: %+v", tspconfigPath, err))
```

Comment:
Invalid tspconfig.yaml provided and refer to the sample file: https://aka.ms/azsdk/tspconfig-sample-mpg to fix the content.